lib/semian/redis.rb

```diff
@@ -15,17 +15,19 @@ def initialize(semian_identifier, *args)
   ResourceBusyError = Class.new(SemianError)
   CircuitOpenError = Class.new(SemianError)
 
-  attr_reader :semian_resource
-
   alias_method :_original_initialize, :initialize
 
   def initialize(*args, &block)
     _original_initialize(*args, &block)
 
-    # This alias is necessary because during a `pipelined` block
-    # the client is replaced by an instance of `Redis::Pipeline` and there is
-    # no way to access the original client.
-    @semian_resource = client.semian_resource
+    # This reference is necessary because during a `pipelined` block the client
+    # is replaced by an instance of `Redis::Pipeline` and there is no way to
+    # access the original client which references the Semian resource.
+    @original_client = client
+  end
+
+  def semian_resource
```

Comment:
If you instantiate the Redis clients (e.g. in your environment in `config/environments/*.rb` in Rails) previously we'd call `#initialize` and call `client.semian_resource` which would eventually call [`Semian.retrieve_or_register`](https://github.com/Shopify/semian/blob/master/lib/semian/adapter.rb#L21), registering the resource.

This solves the problem by caching the client, not the resource—so that `.retrieve_or_register` is not called through `client.resource` until  the first I/O is performed (`#connect` or `#io`).